### ISSUE TITLE
Cursor tracking not working with screen readers

### ISSUE THREAD

**Issue by ndarilek** [mawww/kakoune#1337]
I'm a blind Linux user attempting to use Kakoune with my screen reader. Thus far I've tried Orca, the default GNOME screen reader, and a console screen reader called TDSR. In theory, a console app like Kakoune should pose few problems. I'm having one significant issue, though. The screen reader can't track the cursor. So for instance, if I run kak, then press `a`, then type "This is a test.", I'd expect pressing left-arrow to speak ".", "t", "s", etc. but it doesn't. Let's ignore the fact that I'm not using VI-style movement commands, which don't play well with screen readers anyway.

I've noticed something else. When I review the screen, asking my screen reader to speak the current/'previous/next line, it seems to think I'm on the status line rather than on my position in the text I'm editing.

I don't know a whole lot about how terminals work, so I don't know what to ask for. I do know that Vim, Neovim, Pico, and just about every other console editor I've run does track the cursor just fine, so there's something Kak is doing that fails to work in multiple screen readers. Is there anything unique in how Kak represents its cursor differently than all these other text editors? Maybe some ANSI escape sequences that aren't being emitted, or maybe whatever redraws the status line tricks the cursor into thinking it's on the status line itself and not actually in the text?

Happy to provide additional information or to test proposed fixes. Thanks! I'd really like to try using Kakoune, but this minor issue makes it pretty much a non-starter for me.

**Comment by lenormf** [comment]
Hi,

There's been a couple issues with cursor placement in general, I think the last comment in #1318 gives us a hint: the rendering is controlled completely by the software, and not by the "hardware", which your screen reader probably relies upon (logically).

I wouldn't know how to fix this myself, but it's a good candidate for an explanation as to why other editors pose no problems (they probably handle this "hardware" cursor better).

**Comment by ndarilek** [comment]
Thanks for the pointer! Building from master seems to have solved the problem. I'm using the Fedora package, which doesn't seem to include this commit. There are still odd issues, but I can't quite describe them (it's about 2 AM here and I'm tired :) and they're not showstoppers in the way this one was.

**Comment by lenormf** [comment]
Sure, feel free to create more issues if some things can be improved to allow you use the editor with your tools.

**Comment by mawww** [comment]
Hello, thanks to this issue we have an even stronger case for placing the hardware terminal cursor at the main cursor position even though we dont use it for display.
